tools/explorer/lib/ontology_wrapper.py

```diff
@@ -105,58 +106,136 @@ def GetFieldsForTypeName(
+    if tr <= 0:
+      match_score = ((ma - e) / c) / 2.0
+    else:
+      match_score = (((ma - e) / c) + ((mr - a) / tr)) / 2.0
+
+    return match_score
+
+  def _CreateMatch(self, field_list: List[StandardField],
                    entity_type: EntityType) -> Match:
-    """Determines the match between: EntityType and EntityTypeFields.
+    """Creates a Match instance for an EntityType object and a list of
+    StandardField objects.
 
     calls _CalculateMatchWeight() on field_list and the set of fields belonging
-    to entity_type. The weight function outputs a weight signifying the
-    closeness of the match and an instance of the Match class is created with
-    field_list, entity_type, and weight as arguments.
+    to entity_type. The scoring function outputs a float signifying the
+    closeness of the match, and an instance of the Match class is created with
+    field_list, entity_type, and match_score as arguments.
 
     Args:
-      field_list: A list of EntityTypeField objects.
+      field_list: A list of EntityTypeField objects for a concrete entity.
       entity_type: An EntityType object.
 
     Returns:
-      An instance of the Match class.
+      An instance of Match class.
     """
-    pass
+    canonical_field_set = set()
+    for qualified_field in entity_type.GetAllFields().values():
+      new_entity_type_field = EntityTypeField(
+          qualified_field.field.namespace,
+          qualified_field.field.field,
+          qualified_field.optional,
+          qualified_field.field.increment
+      )
+      canonical_field_set.add(new_entity_type_field)
+
+    match_score = self._CalculateMatchScore(
+        concrete_fields=frozenset(field_list),
+        canonical_fields=frozenset(canonical_field_set)
+    )
+    new_match = Match(
+        field_list,
+        entity_type,
+        match_score
+    )
+    return new_match
 
   def GetEntityTypesFromFields(self,
-                               field_list: List[EntityTypeField],
-                               general_type: str = None) -> List[EntityType]:
-    """Get a list of EntityType objects matching a list of EntityTypeFields.
+                               field_list: List[StandardField],
+                               general_type: str = None,
+                               best_fit: bool = False) -> List[Match]:
+    """Get a list of Match objects containg information on a strength of a
+    match between all entity types defined in DBO and a list of concrete
+    fields.
 
     Args:
-        field_list: a list of EntityTypeField objects to match to an entity
-        general_type: a string indicating a general type name to filter return
-          results.
+      field_list: A list of StandardField objects to match to an entity
+      general_type: A string indicating a general type name to filter return
+        results.
+      best_fit: If True, then return a list of Match objects whose score is
```

Comment:
I think it would be better if the user can specify a maximum number of matches to return or maybe their own score threshold rather than rely on a hard-coded threshold. If the user is going to see or provide score values, I think a [0, 1] or [0, 100] range would be more intuitive than [-1, 1].

Reply:
Done.